### PR TITLE
Refactory of `next_slot` method

### PR DESCRIPTION
**Pull request by davxy**
This PR:
- Simplifies the implementation of `next_slot` method
- Prevents slot worker to exit if inherent data provider creation fails.

Just an idea... maybe we can set an upper bound to the number of allowed failures?

---

Fixes warp sync test 0002 in https://github.com/paritytech/substrate/pull/13078

### CODE REVIEW

File: client/consensus/slots/src/slots.rs [hunk 1]
```diff
@@ -124,24 +124,20 @@ where
 	/// Returns a future that fires when the next slot starts.
 	pub async fn next_slot(&mut self) -> Result<SlotInfo<Block>, Error> {
```
Comment (bkchr) [inline client/consensus/slots/src/slots.rs:125]: This doesn't need to return `Result` anymore. We don't return any error somewhere.

**Review by bkchr — APPROVED** [review]

**Review by melekes — APPROVED** [review]
👍

File: client/consensus/slots/src/slots.rs [hunk 2]
```diff
@@ -150,30 +146,41 @@ where
...
 				.create_inherent_data_providers
 				.create_inherent_data_providers(chain_head.hash(), ())
-				.await?;
+				.await
+			{
+				Ok(x) => x,
+				Err(e) => {
+					log::warn!(
+						target: LOG_TARGET,
+						"Unable to author block in slot. Failure creating inherent data provider: {}",
```
Comment (koute) [inline client/consensus/slots/src/slots.rs:163]: It'd be nice to print out which exact slot it is.

**Review by michalkucharczyk — APPROVED** [review]

**Review by koute — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2278930

**Pull request merged**